core/bin/external_node/src/main.rs

```diff
@@ -117,6 +130,23 @@ async fn init_tasks(
         config.optional.miniblock_seal_queue_capacity,
     );
     task_handles.push(tokio::spawn(miniblock_sealer.run()));
+    let pool = connection_pool.clone();
+    task_handles.push(tokio::spawn(async move {
+        loop {
+            let protocol_version = pool
+                .access_storage()
+                .await
+                .unwrap()
+                .protocol_versions_dal()
+                .last_version_id()
```

Comment:
For main node `SELECT MAX(id) FROM protocol_versions != SELECT protocol_version FROM l1_batches ORDER BY number DESC LIMIT 1` as upgrades are scheduled so there can be a version in `protocol_version` that will be used in future. Though, for EN we currently don't populate `protocol_versions` in advance, I don't think we should assume it here.
i.e. I propose to change `SELECT MAX(id) FROM protocol_versions` to ` SELECT protocol_version FROM l1_batches ORDER BY number DESC LIMIT 1` here